angular/src/app/modules/delivery-management/delivery/available-resource-tab/will-pool/will-pool.component.ts

```diff
@@ -0,0 +1,386 @@
+        this.selectUserType.close();
+        break;
+    }
+    this.pageNumber = 1;
+    this.refresh();
+  }
+
+  onSelectChangeBranch(id: number): void {
+    this.selectedBranchIdsCr = this.selectedBranchIds = this.onSelectChange(this.selectedBranchIdsCr, id);
+    this.listBranchs = this.orderList(this.listBranchs, this.selectedBranchIds);
+  }
+
+  getAllUserTypes(): void {
+    this.listUserTypes = Object.entries(this.APP_ENUM.UserTypeTabAllResource).map((item) => {
+      return {
+        displayName: item[0],
+        value: item[1],
+      };
+    });
+    const listUserTypeValues = this.listUserTypes.map(item => item.value);
+    this.selectedUserTypes = this.selectedUserTypesOld = this.selectedUserTypesCr = this.listUserTypesId = listUserTypeValues;
+  }
+
+  onSelectChangeUserType(id: number): void {
+    this.selectedUserTypesCr = this.selectedUserTypes = this.onSelectChange(this.selectedUserTypesCr, id);
+    this.listUserTypes = this.orderList(this.listUserTypes, this.selectedUserTypes);
+  }
+
+  onSelectChange(listSelect: number[], id: number): number[] {
+    if (listSelect.includes(id)) {
+      return listSelect.filter(res => res != id);
+    }
+    else {
+      listSelect.push(id);
+      return listSelect;
+    }
+  }
+
+  orderList(listAll: any[], listIdSelect: number[]): any[] {
+    const selectedSet = new Set(listIdSelect);
+    const listSelect: any[] = [];
+    const listUnSelect: any[] = [];
+    for (const item of listAll) {
+      if (selectedSet.has(item.id)) {
+        listSelect.push(item);
+      } else {
+        listUnSelect.push(item);
+      }
+    }
+    return [...listSelect, ...listUnSelect];
+  }
+
+  onClickSortDirection(field: number): void {
+    this.fieldSortDirection[field] = !this.fieldSortDirection[field];
+    switch(field) {
+      case this.sortProperties.Contribute:
+        this.sortDataByTotalContribute(this.fieldSortDirection[this.sortProperties.Contribute]);
+        break;
+      case this.sortProperties.EndChargeDate:
+        this.sortDataByEndChargeDate(this.fieldSortDirection[this.sortProperties.EndChargeDate]);
+        break;
+    }
+  }
+
+  sortDataByTotalContribute(direction: boolean): void {
+    this.listWillPool = this.listWillPool.sort((a, b) => {
+      return direction
+        ? a.totalContribute - b.totalContribute
+        : b.totalContribute - a.totalContribute;
+    });
+  }
+
+  sortDataByEndChargeDate(direction: boolean): void {
+    const defaultGetTime = 1;
+    this.listWillPool.forEach(item => {
+      item.accounts.sort((a, b) => {
+        const getValidTimestamp = (dateString: string): number => {
+          const date = new Date(dateString);
+          return Number.isNaN(date.getTime()) ? defaultGetTime : date.getTime();
+        };
+        const dateA = getValidTimestamp(a.endChargeDate);
+        const dateB = getValidTimestamp(b.endChargeDate);
+        return direction 
+          ? dateA - dateB
+          : dateB - dateA;
+      })
+    })
+  }
```

Comment:
No. My logical is if endChargeDate has empty or null value then default value is 1, so this will exist a min value to compare with different values of endChargeDate. By the way, endChargeDates has value default is 1 always arrange behind endChargeDates has values